File: src/algo/sort/BubbleSort.java
package algo.sort;

import java.util.Arrays;
import java.util.List;

/**
 * @author SYAM K
 */
public class BubbleSort {
    public static void main(String[] args) {
        List<Integer> integerList = Arrays.asList(22, 33, 11, 55, 99, 77, 66, 88);
        performBubbleSort(integerList);
        System.out.println(integerList);
    }

    private static void performBubbleSort(List<Integer> integerList) {
        boolean swapPerformed;
        for (int i = 0; i < integerList.size(); i++) {
            swapPerformed = false;
            for (int j = 0; j < integerList.size() - i - 1; j++) {
                if (integerList.get(j) > integerList.get(j + 1)) {
                    swap(integerList, j, j + 1);
                    swapPerformed = true;
                }
            }
            if (!swapPerformed) break;
        }
    }

    public static void swap(List<Integer> integerList, int i, int j) {
        if (i != j && !integerList.get(i).equals(integerList.get(j))) {
            integerList.set(i, integerList.get(i) + integerList.get(j));
            integerList.set(j, integerList.get(i) - integerList.get(j));
            integerList.set(i, integerList.get(i) - integerList.get(j));
        }
    }
}

File: src/algo/sort/BubbleSort.kt
package algo.sort

/**
 * @author SYAM K
 */
fun main() {
    val integerList = mutableListOf(22, 33, 11, 55, 99, 77, 66, 88)
    performBubbleSort(integerList)
    println(integerList)
}

private fun performBubbleSort(integerList: MutableList<Int>) {
    var swapPerformed: Boolean
    for (i in 0 until integerList.size) {
        swapPerformed = false
        for (j in 0 until integerList.size - i - 1) {
            if (integerList[j] > integerList[j + 1]) {
                swap(integerList = integerList, i = j, j = j + 1)
                swapPerformed = true
            }
        }
        if (!swapPerformed) break
    }
}

private fun swap(integerList: MutableList<Int>, i: Int, j: Int) {
    if (i != j && integerList[i] != integerList[j]) {
        integerList[i] = integerList[i] + integerList[j]
        integerList[j] = integerList[i] - integerList[j]
        integerList[i] = integerList[i] - integerList[j]
    }
}
